lib/util/parse-connection.js

```diff
@@ -1,20 +1,31 @@
-const url = require('url');
 const { parse } = require('pg-connection-string');
 const parsePG = parse;
 const isWindows = process && process.platform && process.platform === 'win32';
 
+/**
+ * @param str
+ * @returns {URL}
+ */
+function tryParse(str) {
+  try {
+    return new URL(str);
+  } catch (e) {
+    return null;
+  }
+}
+
 module.exports = function parseConnectionString(str) {
-  const parsed = url.parse(str, true);
-  let { protocol } = parsed;
-  const isDriveLetter = isWindows && protocol && protocol.length === 2;
```

Comment:
protocol is always string, so && protocol && removed

Reply:
Here is the reason why it was made defensive: #3743
It was failing under certain circumstances.

Reply:
This check now not actual, because url.parse can return protocol null, and in URL case this is impossible